frontend/catalyst/jax_extras/tracing.py

```diff
@@ -319,6 +322,49 @@ def __init__(self, boxid: int, e: JaxprEqn):
     return [b.e for b in stable_toposort(boxes)]  # [4]
 
 
+def jaxpr_pad_consts(jaxprs: List[Jaxpr]) -> List[ClosedJaxpr]:
+    """Align the constants of Jaxpr programs. Return the list of corresponding programs accepting
+    same constants."""
+    newvar = gensym(jaxprs, suffix="_")
+
+    # List of constant variables of all jaxprs, preprended with '_'
+    all_mangled_constvars: List[List[Var]] = []
```

Comment:
We always did this, but in the previous version we used the [slightly modified version](https://github.com/PennyLaneAI/catalyst/blob/6f6a74b92d1c630e43f067b8aeb713c73a0b015d/frontend/catalyst/jax_extras/tracing.py#L482) of the Jax function `initial_style_jaxprs_with_common_consts`. Unfortunately, it did not work well with dynamic shapes, so I have replaced it with the new function. The code became much smaller because the original function also managed AbstractRefs - something that we do not need

Reply:
Not for now, feel free to ignore this comment: maybe we should switch to using StableHLO's switch https://github.com/openxla/stablehlo/blob/main/docs/spec.md#case if that is what JAX is defining then JAX probably has a good lowering for them.

Reply:
I really hope that Github will store these properly